### PR TITLE
feature/cplp-832-appdetail-refactoring

### PR DESCRIPTION
**Pull request by qxz2mqe**
https://jira.catena-x.net/browse/CPLP-832

### CODE REVIEW

**Review by ntruchsess — CHANGES_REQUESTED** [review]
It appears your PR contains many files that are completely unrelated to your change containing changes that apparently have been applied to main recently. That also causes quite a lot of conflicts in locations I'm pretty sure you did not really work on. Did you merge main squashing the requests?
Please resolve, I'm not going to review such a mess.
(The easiest was is to start a new branch an pick the commits you intend to keep - that will take way less time than trying to resolve the tons of conflicts you are facing now).

**Review by ntruchsess — CHANGES_REQUESTED** [review]

**Review by ntruchsess — CHANGES_REQUESTED** [review]

**Review by ntruchsess — CHANGES_REQUESTED** [review]
Please be aware that I did rework my comments multiple times to come up with the appropriate suggestion. Please re-check my comments.

File: src/portalbackend/PortalBackend.DBAccess/Repositories/OfferRepository.cs [hunk 1]
```diff
@@ -262,10 +262,9 @@ public Offer CreateOffer(string provider, OfferTypeId offerType, Action<Offer>?
                     a.ProviderCompany!.Name,
                     a.UseCases.Select(uc => uc.Name),
                     a.OfferDescriptions.Select(description => new OfferDescriptionData(description.LanguageShortName, description.DescriptionLong, description.DescriptionShort)),
-                    a.ConsentAssignedOffers.Select(consentAssignedOffer => new OfferAgreement(
-                        consentAssignedOffer.Consent!.AgreementId,
-                        consentAssignedOffer.Consent.Agreement!.Name,
-                        consentAssignedOffer.Consent.ConsentStatusId)),
+                    a.AgreementAssignedOffers.Where(agreementtAssignedOffer=>agreementtAssignedOffer.Agreement!.AgreementAssignedOfferTypes
```
[the remark this answers is not in the document]
Comment (qxz2mqe) [reply src/portalbackend/PortalBackend.DBAccess/Repositories/OfferRepository.cs]: Deleted this method

File: src/portalbackend/PortalBackend.DBAccess/Repositories/OfferRepository.cs [hunk 2]
```diff
@@ -262,10 +262,9 @@ public Offer CreateOffer(string provider, OfferTypeId offerType, Action<Offer>?
...
+                    .Any(aaot => aaot.OfferTypeId == offerTypeId))
+                    .Select(aaot => new AgreementAssignedOfferData(aaot.AgreementId, aaot.Agreement!.Name)).Distinct(),
```
[the remark this answers is not in the document]
Comment (qxz2mqe) [reply src/portalbackend/PortalBackend.DBAccess/Repositories/OfferRepository.cs]: fixed

File: src/marketplace/Offers.Library/Service/OfferService.cs [hunk 3]
```diff
@@ -326,14 +326,24 @@ public async Task<OfferProviderResponse> GetProviderOfferDetailsForStatusAsync(G
         {
             throw new ForbiddenException($"userId {userId} is not associated with provider-company of offer {offerId}");
         }
+        var offerAgreementData = new List<OfferAgreement>();
+        foreach(var agreementConsentData in offerDetail.OfferProviderData.Agreements)
```
Comment (ntruchsess) [inline src/marketplace/Offers.Library/Service/OfferService.cs:330]: ```var offerAgreementData = offerDetail.OfferProviderData.Agreements.Select(a => new OfferAgreement(a.Id,a.Name,a.ConsentStatus == null ? string.Empty : a.ConsentStatus.ToString()));````

**Review by ntruchsess — CHANGES_REQUESTED** [review]

**Review by ntruchsess — APPROVED** [review]

**Pull request merged**